src/pages/list-view-page.js

```diff
@@ -186,232 +192,105 @@ function ListViewPage() {
       });
   }, []);
 
+  function handlePrice(event) {
+    event.preventDefault();
+
+    const {name, value} = event.target;
+    setPrices({ ...prices, [name.replace("-amount","")]: value });
+  }
+
+  function handleArea(event) {
+    event.preventDefault();
+
+    const {name, value} = event.target;
+    setAreas({ ...areas, [name.replace("-amount","")]: value });
+  }
+
+  function handleDone(event){
+    event.preventDefault();
+
+    const id = event.target.id
+    if(id==='prices') setFilters({...filters, [id]: prices});
+    if(id === "bathsbeds") setFilters({...filters, "beds": beds, "baths":baths});
+    if (id === "more") setFilters({ ...filters, "petAllowed": petAllowed, "areas": areas })
+    if (id === "propertyType") setFilters({ ...filters, "propertyType": [house, apartment]})
+
+    setShowFilters({
+      ...showFilters,
+      more: false,
+      price: false,
+      bedBath: false,
+      propertyType: false
+    });
+
+  }
+
+  function handleCheck(event) {
+    const id = event.target.id
+    if (id === "check-pets-allowed") setPetAllowed(event.target.checked);
+    if (id === "check-house") setHouse(event.target.checked);
+    if (id === "check-apartment") setApartment(event.target.checked);
+  }
+
+  function handleOperationType(event){
+    const id = event.target.id
+
+    if (id === "both") {
+      setBoth(!both)
+      setBuy(!both);
+      setRent(!both);
+    };
+    if (id === "check-buy") setBuy(!buy);
+    if (id === "check-rent") setRent(!rent);
+  }
+
+  function handleQuery(event) {
+    setQuery(event.target.value);
+  }
+
+  useEffect(()=>{
+    setFilters({...filters, "operationType":[buy, rent]})
+  },[buy, rent, both])
+
+  useEffect(()=>{
+    setFilters({...filters, "search":query})
+  },[query])
```

Comment:
Because you're not using the same `useEffect` here, I understand that if it still changes the `query` I want to keep the other filters selected.

That is why I am left with the emptiness of why 2 `useEffects`